justfile

```diff
@@ -150,8 +150,55 @@ run: devenv
     $BIN/python manage.py runserver
 
 
-# update npm deps, build payload, and collect for Django
-rebuild-static:
+# Remove built assets and collected static files
+assets-clean:
+    rm -rf assets/dist
+    rm -rf staticfiles
+
+
+# Install the Node.js dependencies
+assets-install:
+    #!/usr/bin/env bash
+    set -eu
+
+    # exit if lock file has not changed since we installed them. -nt == "newer than",
+    # but we negate with || to avoid error exit code
+    test package-lock.json -nt node_modules/.written || exit 0
+
     npm ci
+    touch node_modules/.written
+
+
+# Build the Node.js assets
+assets-build:
+    #!/usr/bin/env bash
+    set -eu
+
+    # get most recently modified src file
+    newest=$(find assets/src -type f | xargs ls -tr | tail -n 1)
+
+    # exit if most recently modified file in assets/src has not changed since
+    # we built the assets. -nt == "newer than", but we negate with || to avoid
+    # error exit code
+    test $newest -nt assets/dist/.written || exit 0
```

Comment:
I think this might be simplified with single `find`:

`find assets/src -type f -newer assets/dist/.written | grep -q . || exit 0`

This will exit with a 1 if we find no files that are newer than .written, and with 0 if we do.